Tests/AsyncHTTPClientTests/HTTPClientRequestTests.swift

```diff
@@ -0,0 +1,444 @@
+            guard let preparedRequest = preparedRequest else { return }
+
+            XCTAssertEqual(preparedRequest.poolKey, .init(
+                scheme: .http,
+                connectionTarget: .domain(name: "example.com", port: 80),
+                tlsConfiguration: nil
+            ))
+            XCTAssertEqual(preparedRequest.head, .init(
+                version: .http1_1,
+                method: .POST,
+                uri: "/post",
+                headers: [
+                    "host": "example.com",
+                    "content-length": "0",
+                ]
+            ))
+            XCTAssertEqual(preparedRequest.requestFramingMetadata, .init(
+                connectionClose: false,
+                body: .fixedSize(0)
+            ))
+
+            guard let buffer = await XCTAssertNoThrowWithResult(try await preparedRequest.body.read()) else { return }
+            XCTAssertEqual(buffer, ByteBuffer())
+        }
+        #endif
+    }
+
+    func testPostWithByteBuffer() {
+        #if compiler(>=5.5) && canImport(_Concurrency)
+        guard #available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *) else { return }
+        XCTAsyncTest {
+            var request = Request(url: "http://example.com/post")
+            request.method = .POST
+            request.body = .byteBuffer(.init(string: "post body"))
+            var preparedRequest: PreparedRequest?
+            XCTAssertNoThrow(preparedRequest = try PreparedRequest(request))
+            guard let preparedRequest = preparedRequest else { return }
+
+            XCTAssertEqual(preparedRequest.poolKey, .init(
+                scheme: .http,
+                connectionTarget: .domain(name: "example.com", port: 80),
+                tlsConfiguration: nil
+            ))
+            XCTAssertEqual(preparedRequest.head, .init(
+                version: .http1_1,
+                method: .POST,
+                uri: "/post",
+                headers: [
+                    "host": "example.com",
+                    "content-length": "9",
+                ]
+            ))
+            XCTAssertEqual(preparedRequest.requestFramingMetadata, .init(
+                connectionClose: false,
+                body: .fixedSize(9)
+            ))
+            guard let buffer = await XCTAssertNoThrowWithResult(try await preparedRequest.body.read()) else { return }
+            XCTAssertEqual(buffer, .init(string: "post body"))
+        }
+        #endif
+    }
+
+    func testPostWithSequenceOfUnknownLength() {
+        #if compiler(>=5.5) && canImport(_Concurrency)
+        guard #available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *) else { return }
+        XCTAsyncTest {
+            var request = Request(url: "http://example.com/post")
+            request.method = .POST
+            let sequence = AnySequence(ByteBuffer(string: "post body").readableBytesView)
+            request.body = .bytes(sequence)
```

Comment:
Do we want to have this overload on `.bytes()`? I thought we only had `.bytes(length: Int, Sequence)` and `.bytes(Collection)`? Do we maybe have a default property on length? If so I think we should remove it.

Reply:
We currently default to `nil` but I was thinking that we should make `length` non-optional and probably also move it the end. WDYT? However, I would then do that in a follow up PR.